### PR TITLE
Add slack plugin

### PR DESCRIPTION
**Pull request by AIFlowML**
# Add Slack Integration for Eliza AI Agent

## Overview
This PR adds a new Slack client integration for the Eliza AI agent, enabling seamless communication between Slack workspaces and the Eliza AI system. The integration includes robust message handling, conversation management, and standalone example scripts for easy testing and deployment.

## Features
- Full Slack bot integration with event handling and message management
- Message deduplication system to prevent duplicate responses
- Thread-aware conversation handling
- Support for attachments and media files
- Conversation summarization capabilities
- Standalone example scripts for testing and development

## Example Scripts
The package includes several standalone scripts in `packages/client-slack/src/examples/`:

1. **standalone-example.ts**
   - Basic bot setup and connection
   - Demonstrates message sending, threading, and image attachments
   - Includes error handling and port management

2. **standalone-summarize.ts**
   - Shows conversation summarization capabilities
   - Demonstrates handling attachments and threaded discussions

3. **standalone-attachment.ts**
   - Comprehensive attachment handling
   - File upload and sharing capabilities

### 1. Slack App Creation
Slack App Creation
- Create a new Slack app using the provided manifest
- Configure bot permissions and event subscriptions

### 2. Bot Configuration
Bot Configuration
- Set up OAuth scopes and bot token
- Configure event subscriptions and interactive components

### Running Example Scripts
1. Set up environment variables in `.env`:
   ```bash
   SLACK_APP_ID=your_app_id
   SLACK_CLIENT_ID=your_client_id
   SLACK_CLIENT_SECRET=your_client_secret
   SLACK_SIGNING_SECRET=your_signing_secret
   SLACK_BOT_TOKEN=xoxb-your-bot-token
   SLACK_VERIFICATION_TOKEN=your_verification_token
   SLACK_CHANNEL_ID=test_channel_id
   ```

2. Run the basic example:
   ```bash
   ts-node packages/client-slack/src/examples/standalone-example.ts
   ```

3. Test summarization:
   ```bash
   ts-node packages/client-slack/src/examples/standalone-summarize.ts
   ```

4. Try attachment handling:
   ```bash
   ts-node packages/client-slack/src/examples/standalone-attachment.ts
   ```

### Verifying Integration
1. Start a new conversation with the bot
2. Test direct mentions and thread replies
3. Try uploading files and requesting summaries
4. Verify message handling in different channel types (DM, public, private)

## Technical Details
- TypeScript-based implementation with full type safety
- Event-driven architecture for message handling
- Built-in error handling and logging
- Configurable message deduplication
- Extensible provider system for easy customization

## Documentation
- Full setup guide in README.md
- Example scripts with inline documentation
- Type definitions for all public APIs
- Environment configuration guide

## Checklist
- [x] TypeScript implementation with full type coverage
- [x] Comprehensive example scripts
- [x] Setup screenshots and documentation
- [x] Error handling and logging
- [x] Message deduplication system
- [x] Thread awareness
- [x] Attachment handling
- [x] Environment validation
- [x] Testing instructions

### CODE REVIEW

**Comment by CREWorx** [comment]
Very useful integration!

File: docs/api/classes/AgentRuntime.md
Comment (sirkitree) [inline docs/api/classes/AgentRuntime.md:1]: These docs changes shouldn't be happening.

File: docs/api/classes/CacheManager.md
Comment (sirkitree) [inline docs/api/classes/CacheManager.md:1]: Nor should these docs changes be happening.

File: docs/api/classes/DbCacheAdapter.md
Comment (sirkitree) [inline docs/api/classes/DbCacheAdapter.md:1]: here as well.

**Review by sirkitree — CHANGES_REQUESTED** [review]
overall this appears to have some really great functionality, looking forward to trying it out further. 

Howver there are some path changes in the /docs/ that you probably did not intend to happen (i've seen this happen in other PRs too). Please double check those as I think you'll want to remove them.

**Comment by lalalune** [comment]
Merged on `develop`

**Pull request merged**